File: AcademicPlanFrontend/src/app/service/course-project.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BaseService} from './base.service';
import {CourseProject} from '../entity/course-project';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})

export class CourseProjectService extends BaseService<CourseProject>{

  constructor(private http: HttpClient){
    super(http, "/courseProject");
  }

  getByExplanatoryNoteId(explanatoryNoteId: number): Observable<CourseProject>{
    return this.http.get<CourseProject>('/api/courseProject/byExplanatoryNoteId/' + explanatoryNoteId);
  }
}

File: AcademicPlanBackend/src/main/java/com/bntu/fitr/poit/zholudev/diplom/service/CourseProjectService.java
package com.bntu.fitr.poit.zholudev.diplom.service;

import com.bntu.fitr.poit.zholudev.diplom.entity.CourseProject;

public interface CourseProjectService extends EntityService<CourseProject> {
    CourseProject getByExplanatoryNoteId(Long explanatoryNoteId);
}
